test/system_SUITE.erl

```diff
@@ -160,6 +186,30 @@ test_successful_connection_with_a_full_permission_token_and_explicitly_configure
         amqp_channel:call(Ch, #'queue.declare'{exclusive = true}),
     close_connection_and_channel(Conn, Ch).
 
+test_successful_token_refresh(Config) ->
+    Duration = 5,
+    {_Algo, Token} = generate_expirable_token(Config, [<<"rabbitmq.configure:vhost1/*">>,
+                                                       <<"rabbitmq.write:vhost1/*">>,
+                                                       <<"rabbitmq.read:vhost1/*">>],
+                                                       Duration),
+    Conn     = open_unmanaged_connection(Config, 0, <<"vhost1">>, <<"username">>, Token),
+    {ok, Ch} = amqp_connection:open_channel(Conn),
+
+    {_Algo, Token2} = generate_valid_token(Config, [<<"rabbitmq.configure:vhost1/*">>,
+                                                    <<"rabbitmq.write:vhost1/*">>,
+                                                    <<"rabbitmq.read:vhost1/*">>]),
+    ?UTIL_MOD:wait_for_token_to_expire(Duration),
```

Comment:
Waiting time is in milliseconds, so it should be `?UTIL_MOD:wait_for_token_to_expire(Duration * 1000)`. This combined with the longer expiration time above would make the test pass even if the `update_secret` is not called.